skins/oasis/modules/LatestActivityController.class.php

```diff
@@ -1,110 +1,87 @@
+class LatestActivityController extends WikiaController {
+	const MAX_ELEMENTS = 4;
 
 	public function executeIndex() {
-		wfProfileIn(__METHOD__);
-		$maxElements = 4;
-
-		global $wgLang, $wgContentNamespaces, $wgMemc, $wgUser;
-		$this->moduleHeader = wfMsg('oasis-activity-header');
-		
-		if( empty($this->userName) ) {
-			$mKey = wfMemcKey('mOasisLatestActivity');
-			$feedData = $wgMemc->get($mKey);
-		}
-		
-		if (empty($feedData)) {
+		global $wgLang, $wgContentNamespaces, $wgMemc;
+
+		$mKey = wfMemcKey( 'mOasisLatestActivity' );
+		$feedData = $wgMemc->get( $mKey );
+
+		if ( empty( $feedData ) ) {
 			// data provider
-			$includeNamespaces = implode('|', $wgContentNamespaces);
-			$parameters = array(
+			$includeNamespaces = implode( '|', $wgContentNamespaces );
+			$parameters = [
 				'type' => 'widget',
-				//'tagid' => $id,
-				'maxElements' => $maxElements,
-				'flags' => array('shortlist'),
+				'maxElements' => self::MAX_ELEMENTS,
+				'flags' => [ 'shortlist' ],
 				'uselang' => $wgLang->getCode(),
 				'includeNamespaces' => $includeNamespaces
-			);
-			
-			$feedProxy = new ActivityFeedAPIProxy($includeNamespaces, $this->userName);
-			$feedProvider = new DataFeedProvider($feedProxy, 1, $parameters);
-			$feedData = $feedProvider->get($maxElements);
-			if( empty($this->userName) ) {
-				$wgMemc->set($mKey, $feedData);
-			}
+			];
+
+			$feedProxy = new ActivityFeedAPIProxy( $includeNamespaces );
+			$feedProvider = new DataFeedProvider( $feedProxy, 1, $parameters );
+			$feedData = $feedProvider->get( self::MAX_ELEMENTS );
+
+			$wgMemc->set( $mKey, $feedData );
 		}
-		
+
 		// Time strings are slow to calculate, but we still want them to update frequently (60 seconds)
-		if( empty($this->userName)) {
-			$mKeyTimes = wfMemcKey('mOasisLatestActivity_times', $wgLang->getCode());
-			$this->changeList = $wgMemc->get($mKeyTimes, array());
-		}
-		
-		if(empty($this->changeList) && !empty($feedData) && is_array($feedData['results'])) {
-			$changeList = array();
-			foreach( $feedData['results'] as $change ) {
-				$item = array();
-				$item['time_ago'] = wfTimeFormatAgoOnlyRecent($change['timestamp']); // TODO: format the timestamp on front-end to allow longer caching in memcache?
-				$item['user_name'] = $change['username'];
-				$item['avatar_url'] = AvatarService::getAvatarUrl( $item['user_name'], 20 );
-				$item['user_profile_url'] = AvatarService::getUrl( $item['user_name'] );
-				$item['user_href'] = AvatarService::renderLink( $item['user_name'] );
+		$mKeyTimes = wfMemcKey( 'mOasisLatestActivity_times', $wgLang->getCode() );
+		$changeList = $wgMemc->get( $mKeyTimes, [] );
+
+		if ( empty( $changeList ) && !empty( $feedData ) && is_array( $feedData['results'] ) ) {
+			foreach ( $feedData['results'] as $change ) {
+				$item = [];
+				$item['time_ago'] = wfTimeFormatAgo( $change['timestamp'] );
+				$item['user_name'] = User::isIP( $change['username'] )
+					? wfMessage( 'oasis-anon-user' )->escaped()
```

Comment:
You should avoid caching data that is dependent on user lang or settings. This can lead us to serving broken or wrong content to end user.